bndtools.core/src/bndtools/launch/BndDependencySourceContainer.java

```diff
@@ -92,8 +92,24 @@ public ISourceContainerType getType() {
                             bundleFile = ResourcesPlugin.getWorkspace().getRoot().getFile(bundlePath);
                         }
                         if (bundleFile != null) {
-                            ArchiveSourceContainer tempArchiveCont = new ArchiveSourceContainer(bundleFile, false);
-                            result.add(tempArchiveCont);
+                            // check to see if this archive came from a repo that encodes the source project name
+                            final String sourceProjectName = runbundle.getAttributes().get("sourceProjectName");
+
+                            if (sourceProjectName != null) {
+                                try {
+                                    IProject sourceProject = ResourcesPlugin.getWorkspace().getRoot().getProject(sourceProjectName);
+
+                                    if (sourceProject.exists()) {
+                                        IJavaProject javaSourceProject = JavaCore.create(sourceProject);
+                                        result.add(new JavaProjectSourceContainer(javaSourceProject));
```

Comment:
Should the default be the archive source container if we fail to find the source project?